pallets/xyk/src/lib.rs

```diff
@@ -619,44 +703,28 @@ impl<T: Config> Pallet<T> {
 		Ok(user_mangata_rewards_amount)
 	}
 
-	pub fn calculate_rewards_amount(
-		user: AccountIdOf<T>,
-		liquidity_asset_id: TokenId,
-		block_number: u32,
-	) -> Result<(Balance, i128), DispatchError> {
-		let current_time = block_number / TIMEBLOCKRATIO;
-		let work_user = Self::calculate_work_user(user.clone(), liquidity_asset_id, current_time)?;
-		let work_pool = Self::calculate_work_pool(liquidity_asset_id, current_time)?;
-
-		let already_claimed_user =
-			LiquidityMiningUserClaimed::<T>::try_get((user, &liquidity_asset_id))
-				.unwrap_or_else(|_| 0 as i128);
-
-		let user_rewards_amount =
-			Self::calculate_rewards(work_user, work_pool, liquidity_asset_id, block_number)?;
-
-		Ok((user_rewards_amount, already_claimed_user))
-	}
-
 	pub fn calculate_work(
 		asymptote: Balance,
 		time: u32,
 		last_checkpoint: u32,
 		cummulative_work_in_last_checkpoint: U256,
 		missing_at_last_checkpoint: U256,
 	) -> Result<U256, DispatchError> {
-		let time_passed = time - last_checkpoint;
+		let time_passed = time
+			.checked_sub(last_checkpoint)
+			.ok_or_else(|| DispatchError::from(Error::<T>::PastTimeCalculation))?;
 
 		let asymptote_u256: U256 = asymptote.into();
 		let cummulative_work_new_max_possible: U256 = asymptote_u256
 			.checked_mul(U256::from(time_passed))
 			.ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
 		let base = missing_at_last_checkpoint
-			.checked_mul(U256::from(11))
-			.ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+			.checked_mul(U256::from(106))
+			.ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))? /
+			U256::from(6);
 
 		let precision: u32 = 10000;
-		let q_pow = Self::calculate_q_pow(1.1, time_passed);
+		let q_pow = Self::calculate_q_pow(Q, time_passed);
```

Comment:
so q_pow will be removed in the future ? (not yet i mean)

Reply:
not in close future, passed to auditor to deep check used pow fn